src/Tasks/Common/Resources/Strings.resx

```diff
@@ -758,4 +758,12 @@ The following are names of parameters or literal values and should not be transl
     <value>NETSDK1163: Cannot emit symbols when publishing for .NET 5 with Crossgen2 using composite mode.</value>
     <comment>{StrBegin="NETSDK1163: "}</comment>
   </data>
-</root>
+  <data name="TrimmingWindowsFormsRequiresCom" xml:space="preserve">
+    <value>NETSDK1164: Trimming Windows Forms is dangerous and requires 'BuiltInComSupport' to be set to 'true'.</value>
```

Comment:
This phrasing seems both too harsh and too mild. 😄 On the one hand, I think "dangerous" probably overstates things -- I didn't think it was any more dangerous than any other trim-incompatible APIs.

On the other hand, we need to be clear that the configuration here is unsupported and not recommended.

My proposal is the following

```suggestion
    <value>NETSDK1164: Windows Forms requires COM, which is not trim-compatible and is disabled by default when trimming. To enable COM and proceed, set `<BuiltInComSupport>true</BuiltInComSupport>`. This configuration is not supported or recommended.</value>
```

Reply:
I don't think the message should speak about COM. There are other things in WinForms which are problematic for trimming - not just COM (although COM is the biggest offender). I think the message should be something like:
```
Windows Forms is not supported with trimming enabled. Please go to https://aka.ms/... for more details.
```

I would introduce a new property which bypasses this warning/error and which gets us back to roughly 5.0 behavior.

If we stick to the current logic around COM:
Some WinForms apps will be broken because of `CustomResourceTypesSupport`, so we should then also ask people to enable that feature switch to get WinForms. And there might be others... we simply don't know because we didn't do the work for WinForms.